handler.go

```diff
@@ -24,6 +25,23 @@ type Request struct {
 	QueryString string
 	Method      string
 	Host        string
+
+	ctx context.Context
+}
+
+// Context returns the request's context.
+func (r *Request) Context() context.Context {
+	return r.ctx
+}
+
+// WithContext changes the request's context.
+func (r *Request) WithContext(ctx context.Context) *Request {
+	if ctx == nil {
```

Comment:
I don't like the panic here, why did we go for this? Is it copied from the Go team's stdlib or personal preference?

Reply:
Yes it was copied from stdlib

Reply:
Seems like stdlib suggest that `WithContext` should return an error, if one can be generated.

I really don't like to see `panic` used to validate variables.


https://golang.org/src/net/http/request.go?s=28311:28412#L831

Reply:
This mimics https://golang.org/src/net/http/request.go?s=28311:28412#L358 pretty close

Reply:
From which part was it copied @SpaWn2KiLl ?

Reply:
It was copied from here https://golang.org/src/net/http/request.go?s=28311:28412#L358